lib/ClangImporter/ClangImporter.cpp

```diff
@@ -4215,9 +4207,22 @@ ClangImporter::getSwiftExplicitModuleDirectCC1Args() const {
 
   if (!Impl.SwiftContext.SearchPathOpts.ScannerPrefixMapper.empty()) {
     // Remap all the paths if requested.
+    // FIXME: Should we also do
+    // https://github.com/swiftlang/llvm-project/pull/10723?
     llvm::PrefixMapper Mapper;
-    clang::tooling::dependencies::DepscanPrefixMapping::configurePrefixMapper(
-        Impl.SwiftContext.SearchPathOpts.ScannerPrefixMapper, Mapper);
+    SmallVector<llvm::MappedPrefix> Prefixes;
+    if (auto E = llvm::MappedPrefix::transformJoined(
+            Impl.SwiftContext.SearchPathOpts.ScannerPrefixMapper, Prefixes)) {
+      // Save this string. In general the diagnostic might outlive this
+      // function.
+      auto identifier =
```

Comment:
Is this an idiom that we use? To save strings are identifiers? I wonder if you wanted to use ASTContext::AllocateCopy instead. 

Otherwise looks good to me.

Reply:
> I wonder if you wanted to use ASTContext::AllocateCopy instead

Yes, good thinking. Thanks!